public/main.js

```diff
@@ -0,0 +1,55 @@
+function doXHR(aURL, aBodyObj, aAlertText, aDebug=true) {
+  return new Promise((resolve) => {
+    let xhr = new XMLHttpRequest();
```

Comment:
Why XMLHttpRequest and not the promise-based [`fetch()` API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Uploading_JSON_data)?

Reply:
Filed issue#18